app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -316,10 +303,40 @@ public void onClick(final SweetAlertDialog sDialog) {
     private void enableBluetooth() {
         final BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
         if (!bluetoothAdapter.isEnabled()) {
-            LOGGER.info("Requesting to enable Bluetooth");
-            final Intent intent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
-            this.startActivityForResult(intent, MainActivity.REQUEST_CODE_ENABLE_BLUETOOTH);
-            LOGGER.info("Request to enable Bluetooth sent");
+            new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
+                    .setTitleText(getString(R.string.enable_bluetooth_questionmark))
+                    .setContentText(getString(R.string.enable_bluetooth_explanation))
+                    .setCancelText(getString(R.string.cancel))
+                    .setConfirmText(getString(R.string.dialog_ok))
+                    .setConfirmClickListener(new SweetAlertDialog.OnSweetClickListener() {
+
+                        /**
+                         * {@inheritDoc}
+                         */
+                        @Override
+                        public void onClick(final SweetAlertDialog sweetAlertDialog) {
+                            sweetAlertDialog.dismiss();
+                            LOGGER.info("Requesting to enable Bluetooth");
+                            final Intent intent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
+                            startActivityForResult(intent, MainActivity.REQUEST_CODE_ENABLE_BLUETOOTH);
```

Comment:
I would suggest to remove the `bluetoothButtonPressed` flag and replace it with additional data in this intent so that `onActivityResult` can detect it with the `data` parameter